docs/src/development/proposals/MEP8/README.md

```diff
@@ -0,0 +1,363 @@
+  Path           *string
+  // Device where the filesystem is created on, must be the full device path seen by the OS
+  Device         Device
+  // Format is the type of filesystem should be created
+  Format         Format
+  // Label is optional enhances readability
+  Label          *string
+  // MountOptions which might be required
+  MountOptions   []MountOption
+  // Options during filesystem creation
+  Options        []FilesystemOption
+}
+
+// Disk represents a single block device visible from the OS, required
+type Disk struct {
+  // Device is the full device path
+  Device          Device
+  // PartitionPrefix specifies which prefix is used if device is partitioned
+  // e.g. device /dev/sda, first partition will be /dev/sda1, prefix is therefore /dev/sda
+  // for nvme drives this is different, the prefix there is typically /dev/nvme0n1p
+  PartitionPrefix string
+  // Partitions to create on this device
+  Partitions      []Partition
+  // WipeOnReinstall, if set to true the whole disk will be erased if reinstall happens
+  // during fresh install all disks are wiped
+  WipeOnReinstall bool
+}
+
+// Raid is optional, if given the devices must match.
+// TODO inherit GPTType from underlay device ?
+type Raid struct {
+  // Name of the raid device, most often this will be /dev/md0 and so forth
+  Name    string
+  // Devices the devices to form a raid device
+  Devices []Device
+  // Level the raidlevel to use, can be one of 0,1,5,10 
+  // TODO what should be support
+  Level   RaidLevel
+  // Options required during raid creation, example: --metadata=1.0 for uefi boot partition
+  Options []RaidOption
+  // Spares defaults to 0
+  Spares  int
+}
+
+// Partition is a single partition on a device, only GPT partition types are supported
+type Partition struct {
+  // Number of this partition, will be added to partitionprefix
+  Number    int
+  // Label to enhance readability
+  Label     *string
+  // Size given in kubernetes resource metrics
+  // if "-1" is given the rest of the device will be used, this requires Number to be the highest in this partition
+  Size      string
+  // GUID of this partition
+  GUID      *GUID
+  // GPTType defines the GPT partition type
+  GPTType   *GPTType
+}
+
+const (
+  // VFAT is used for the UEFI boot partition
+  VFAT = Format("vfat")
+  // EXT3 is usually only used for /boot
+  EXT3 = Format("ext3")
+  // EXT4 is the default fs
+  EXT4 = Format("ext4")
+  // SWAP is for the swap partition
+  SWAP = Format("swap")
+  // None
+  NONE = Format("none")
+
+  // GPTBoot EFI Boot Partition
+  GPTBoot = GPTType("ef00")
+  // GPTLinux Linux Partition
+  GPTLinux = GPTType("8300")
+  // GPTLinuxRaid Linux Raid Partition
+  GPTLinuxRaid = GPTType("fd00")
+  // GPTLinux Linux Partition
+  GPTLinuxLVM = GPTType("8e00")
+  // EFISystemPartition see https://en.wikipedia.org/wiki/EFI_system_partition
+  EFISystemPartition = GUID("C12A7328-F81F-11D2-BA4B-00A0C93EC93B")
```

Comment:
EFISystemPartition and GPTBoot is a redundant information. If the partition type is set to "ef00" (EFI System Partition) then the Disk identifier (GUID) is automatically set to "C12A7328-F81F-11D2-BA4B-00A0C93EC93B", thus it's not necessary to set both.

Reply:
Thanks removed